lib/facter/augeasprovider_grub_version.rb

```diff
@@ -0,0 +1,11 @@
+Facter.add(:augeasprovider_grub_version) do
+  if File.exists?('/etc/default/grub')
+    version = 2
+  elsif File.exists?('/boot/efi/EFI/redhat/grub.conf') || File.exists?('/boot/grub/menu.lst')
+    version = 1
+  else
+    fail('Failed to evaluate grub version')
```

Comment:
What happens in facter when the fact fails? Shouldn't it simply return `nil` instead?